### PR TITLE
expand the doc string of `Union{}`

### PR DESCRIPTION
**Pull request by nsajko**
Give some characterizations/relevant terms from type theory.

Clarify the, previously unexplained, usage of "bottom" in names. Cross-reference the subtyping doc string.

Add an entry to the doctest, clarifying the last sentence.

### CODE REVIEW

File: base/docs/basedocs.jl [hunk 1]
```diff
@@ -3186,13 +3186,23 @@ Any
     Union{}
 
 `Union{}`, the empty [`Union`](@ref) of types, is the type that has no values. That is, it has the defining
-property `isa(x, Union{}) == false` for any `x`. `Base.Bottom` is defined as its alias and the type of `Union{}`
-is `Core.TypeofBottom`.
+property `isa(x, Union{}) == false` for any `x`. Thus it is an *empty*/*uninhabited* type.
+
+A property of `Union{}` is that it's the *bottom* type of the type system. That is, for each `T`
```
Comment (Keno) [inline base/docs/basedocs.jl:3191]: Just say for each `T::Type`?

File: base/docs/basedocs.jl [hunk 2]
```diff
@@ -3186,13 +3186,23 @@ Any
     Union{}
 
 `Union{}`, the empty [`Union`](@ref) of types, is the type that has no values. That is, it has the defining
-property `isa(x, Union{}) == false` for any `x`. `Base.Bottom` is defined as its alias and the type of `Union{}`
-is `Core.TypeofBottom`.
+property `isa(x, Union{}) == false` for any `x`. Thus it is an *empty*/*uninhabited* type.
```
Comment (Keno) [inline base/docs/basedocs.jl:3189]: I would switch the order of these two paragraphs in the docstring. There are other uninhabited types, but `Union{}` is the unique lattice bottom. So put that first an then say: "In particular, this implies..."

File: base/docs/basedocs.jl [hunk 3]
```diff
@@ -3186,13 +3186,23 @@ Any
...
+such that `T isa Type`, `Union{} <: T`. Refer to the subtyping operator's documentation: [`<:`](@ref).
+
+`Base.Bottom` is defined as its alias and the type of `Union{}` is `Core.TypeofBottom`.
 
 # Examples
 ```jldoctest
 julia> isa(nothing, Union{})
 false
+
+julia> Union{} <: Union{} <: Int
```
Comment (Keno) [inline base/docs/basedocs.jl:3201]: I don't know if two `Union{}` in this example make this any clearer. `T<:T` is always true. Adding it here might make it seem like it's a unique property of `Union{}`, which it is not.

File: base/docs/basedocs.jl (hunk 2 above)
Comment (nsajko) [reply base/docs/basedocs.jl]: Is it OK now?

**Review by Keno — APPROVED** [review]

File: base/docs/basedocs.jl [hunk 4]
```diff
@@ -3185,14 +3185,24 @@ Any
...
 # Examples
 ```jldoctest
 julia> isa(nothing, Union{})
 false
+
+julia> Union{} <: Int
+true
+
+julia> typeof(Union{}) === Core.TypeofBottom
+true
```
Comment (Keno) [inline base/docs/basedocs.jl:3205]: Maybe also show
```
julia> isa(Union{}, Union)
false
```
here? But I wonder whether all the discussion of `Bottom` shouldn't go into the extended help section. It's a bit in the weeds.

Comment (Keno) [reply base/docs/basedocs.jl:3205]: In particular, it's something you need to know if you're doing type system manipulation, but it's not really something you need to know if you're just wondering what `Union{}` is.

Comment (nsajko) [reply base/docs/basedocs.jl:3205]: IMO an extended help section isn't necessary for such a tiny doc string?

**Pull request merged**